### PR TITLE
Add firenze.js ORM

### PR DESCRIPTION
**Pull request by fahad19**
* GitHub: https://github.com/fahad19/firenze
* npm: https://www.npmjs.com/package/firenze

#### Notable features:

* Written in ES6
* Works in both node/io.js and the browser
* Adapter based structure to plug in any database/store
* Highly extensible with Behavior pattern
* Promise based workflow
* Strong validation support
* Decent documentation, tests and coverage
* Small footprint of ~30kB minified file

#### Current community contributed adapters:

* [localStorage](https://github.com/fahad19/firenze-adapter-localstorage) (for browser only)
* [Memory](https://github.com/fahad19/firenze-adapter-memory) (works in both node and the browser)
* [MySQL](https://github.com/fahad19/firenze-adapter-mysql)
* [Redis](https://github.com/fahad19/firenze-adapter-redis)
* [SQLite](https://github.com/alexweber/firenze-adapter-sqlite)

More adapters would pop up on npm: https://www.npmjs.com/browse/keyword/firenze-adapter

(A REST adapter is on top of my list at the moment.)

### CODE REVIEW

**Comment by sindresorhus** [comment]
The readme should really outline how it's better than the alternatives. There's just so many alternatives and as a user I have no idea which one to pick.

File: readme.md [hunk 1]
```diff
@@ -450,6 +450,7 @@ Inspired by the [awesome](https://github.com/sindresorhus/awesome) list thing.
 	- [Iridium](https://github.com/SierraSoftworks/Iridium) - A high performance MongoDB ORM with support for promises, distributed caching, preprocessing, validation and plugins.
 	- [OpenRecord](https://github.com/PhilWaldmann/openrecord) - ORM for PostgreSQL, MySQL, SQLite3 and RESTful datastores. Similar to ActiveRecord.
 	- [orm2](https://github.com/dresende/node-orm2) - ORM for PostgreSQL, MariaDB, MySQL, Amazon Redshift, SQLite, MongoDB.
+	- [firenze](https://github.com/fahad19/firenze) - Adapter based ORM for server and the browser. Supports MySQL, Memory, Redis, localStorage and more.
```
Comment (sindresorhus) [inline readme.md]: Should it be `adapter-based`? I also don't think the server/browser mention is relevant seeing how this list is for Node.js.

How about:

> Adapter-based ORM for MySQL, Memory, Redis, localStorage and more.

**Comment by fahad19** [comment]
@sindresorhus: I am assuming people would take notice of the `Key Features` section here: https://github.com/fahad19/firenze#key-features

One core feature that just landed in last version a few minutes ago is [Behavior pattern](https://github.com/fahad19/firenze#behavior) in Models, which I haven't found in any of the other existing ORMs.

**Comment by sindresorhus** [comment]
You don't sell a product by listing its features. You sell it by telling a story that illustrates how it's better. I'll merge, but you didn't sell me on it, and I'm still no wiser on why I would go with Firenze ;)

> Written in ES6

Not really a feature.

**Pull request closed without merging**